haystack/tools/component_tool.py

```diff
@@ -412,6 +412,23 @@ def _create_property_schema(self, python_type: Any, description: str, default: A
                 f"Pydantic models (e.g. {python_type.__name__}) are not supported as input types for "
                 f"component's run method."
             )
+        elif python_type is dict or (origin is dict):
+            schema = {"type": "object", "description": description, "additionalProperties": True}
```

Comment:
@anakin87 do you happen to know if by adding `additionalProperties` here we run the risk of breaking ChatGenerators that don't support it?

Reply:
+1 on the structured output or if you set the JSON response to strict. But that might be a separate issue we need to open where if structured output or strict is set in OpenAIChatGenerator then it should probably go in and update all `additionalProperties` to false.